pkg/sql/catalog/schemaexpr/expr.go

```diff
@@ -74,29 +75,27 @@ func DequalifyAndTypeCheckExpr(
 // tree.TypedExpr would be dangerous. It contains dummyColumns which do not
 // support evaluation and are not useful outside the context of type-checking
 // the expression.
-func DequalifyAndValidateExpr(
+func DequalifyAndValidateExprImpl(
 	ctx context.Context,
-	desc catalog.TableDescriptor,
 	expr tree.Expr,
 	typ *types.T,
 	context string,
 	semaCtx *tree.SemaContext,
 	maxVolatility volatility.V,
 	tn *tree.TableName,
+	getAllNonDropColumnsFn func() colinfo.ResultColumns,
```

Comment:
Would it make sense to have some type of interface, just trying to think how we can make these cleaner in general?

Reply:
It definitely makes sense! I just chosen to not do it in this PR as it's always very bloated.

Reply:
Its okay to leave this a as a follow on